tests/PHPUnit/Integration/Archive/ArchiveStateMetadataTest.php

```diff
@@ -0,0 +1,239 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Tests\Integration\Archive;
+
+use Piwik\Archive;
+use Piwik\Archive\ArchiveState;
+use Piwik\Archive\ArchiveInvalidator;
+use Piwik\Config;
+use Piwik\DataAccess\Model;
+use Piwik\DataTable;
+use Piwik\Date;
+use Piwik\Log\NullLogger;
+use Piwik\Segment;
+use Piwik\Tests\Fixtures\OneVisitorTwoVisits;
+use Piwik\Tests\Framework\TestCase\IntegrationTestCase;
+
+/**
+ * @group ArchiveStateMetadataTest
+ * @group Core
+ */
+class ArchiveStateMetadataTest extends IntegrationTestCase
+{
+    /**
+     * @var OneVisitorTwoVisits
+     */
+    public static $fixture;
+
+    /**
+     * @var string
+     */
+    private $archiveDate;
+
+    /**
+     * @var Segment|null
+     */
+    private $archiveSegment;
+
+    /**
+     * @var int
+     */
+    private $archiveSite;
+
+    /**
+     * @var ArchiveInvalidator
+     */
+    private $invalidator;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->archiveDate = Date::factory(self::$fixture->dateTime)->toString();
+        $this->archiveSite = self::$fixture->idSite;
+
+        $this->invalidator = new ArchiveInvalidator(new Model(), new NullLogger());
+    }
+
+    public function tearDown(): void
+    {
+        parent::tearDown();
+
+        Date::$now = null;
+    }
+
+    /**
+     * @dataProvider archiveStateData
+     *
+     * @param array<string> $pluginsToInvalidate
+     */
+    public function test_getDataTableFromNumeric_returnsArchiveStateInMetadata(
+        int $nowTimestamp,
+        ?string $segment,
+        array $daysToRememberInvalidationsFor,
+        ?array $pluginsToInvalidate,
+        string $expectedInitialArchiveState,
+        string $expectedFinalArchiveState
+    ): void {
+        Date::$now = $nowTimestamp;
+
+        if (null !== $segment) {
+            $this->archiveSegment = new Segment($segment, [$this->archiveSite]);
+        }
+
+        $this->setUpInitialState($expectedInitialArchiveState);
+        $this->invalidateArchives($daysToRememberInvalidationsFor, $pluginsToInvalidate);
+        $this->disableArchiving();
+
+        $dataTable = $this->getDataTable();
+
+        self::assertSame(
+            $expectedFinalArchiveState,
+            $dataTable->getMetadata(DataTable::ARCHIVE_STATE_METADATA_NAME)
+        );
+    }
+
+    /**
+     * @return iterable<string, array>
+     */
+    public function archiveStateData(): iterable
```

Comment:
to more or less keep it in line with existing data provider methods
```suggestion
    public function getArchiveStateTestData(): iterable
```